partner_ref_unique/models/res_partner.py

```diff
@@ -1,38 +1,29 @@
 # Copyright 2016 Antonio Espinosa
+# Copyright 2020 Tecnativa - João Marques
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import _, api, fields, models
+from odoo import _, api, models
 from odoo.exceptions import ValidationError
 
 
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
-    # This related is needed in order to trigger the check when changing the
-    # value on res.company
-    partner_ref_unique = fields.Selection(
-        related="company_id.partner_ref_unique", store=True,
-    )
-
-    @api.multi
-    @api.constrains("ref", "is_company", "company_id", "partner_ref_unique")
+    @api.constrains("ref", "is_company", "company_id")
     def _check_ref(self):
-        for partner in self:
-            mode = partner.partner_ref_unique
-            if partner.ref and (
-                mode == "all" or (mode == "companies" and partner.is_company)
-            ):
+        for partner in self.filtered("ref"):
+            # If the company is not defined, adopt default strategy for user company
```

Comment:
This comment is outdated, you can update it to:
```suggestion
            # If the company is not defined in the partner, take current user company
```